baseapp/util_test.go

```diff
@@ -0,0 +1,64 @@
+package baseapp
+
+import (
+	"github.com/cosmos/cosmos-sdk/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// CheckState is an exported method to be able to access baseapp's
+// checkState in tests.
+//
+// This method is only accessible in baseapp tests.
+func (app *BaseApp) CheckState() *state {
```

Comment:
Are we extending BaseApp functionality here in a test file?

Reply:
Yes, just to access few private methods in test files.

Reply:
Then let's move this out of a test file and into something else. I don't think we should extend type's APIs in test files usually.

Reply:
Agreed, let's do this in a follow-up PR: https://github.com/cosmos/cosmos-sdk/issues/10282.

@atheeshp can you add TODOs in the method comments in this file, poiting to #10282?